src/search.cpp

```diff
@@ -440,10 +442,11 @@ Workload Search::refine(Workload wl) {
     set<TimedSpec> specs = wl.get_all_specs();
     for (set<TimedSpec>::iterator it = specs.begin(); it != specs.end(); it++) {
         TimedSpec tspec = *it;
-        wl_spec_t wspec = tspec.get_wl_spec();
+        WlSpec* wspec = tspec.get_wl_spec();
         // TODO: Generalize to other constructs?
-        if (!holds_alternative<Comp>(wspec)) continue;
-        Comp comp = get<Comp>(wspec);
+        Comp* compSpec = dynamic_cast<Comp*>(wspec);
+        if (!compSpec) continue;
+        Comp comp = *compSpec;
```

Comment:
This is not necessary, you can use the pointer in code.

Reply:
[Search handles pointers more elegantly](https://github.com/all-things-networking/fperf/pull/27/commits/00fe78d47212573296f3abb7df4352785a85d2f3)